subsys/mgmt/smp_bt.c

```diff
@@ -127,16 +127,28 @@ static u16_t smp_bt_get_mtu(const struct net_buf *nb)
 	return mtu - 3;
 }
 
-static void smp_bt_buf_free(struct net_buf *nb)
+static void smp_bt_ud_free(void *ud)
 {
-	struct smp_bt_user_data *ud = net_buf_user_data(nb);
+	struct smp_bt_user_data *user_data = ud;
+
+	if (user_data->conn) {
+		bt_conn_unref(user_data->conn);
+		user_data->conn = NULL;
+	}
+}
+
+static int smp_bt_ud_copy(struct net_buf *dst, const struct net_buf *src)
+{
+	struct smp_bt_user_data *ud;
+
+	ud = net_buf_user_data(dst);
+	memcpy(ud, net_buf_user_data(src), sizeof(src->user_data));
```

Comment:
Since you know that the full user data is of type smp_bt_user_data, I'd avoid this memcpy and just directly do:

```
if (src_ud->conn) {
        dst_ud->conn = bt_conn_ref(src_ud->conn);
}
```

I think that would be the clean way to do this.